Ntuplizer/plugins/Ntuplizer.cc

```diff
@@ -245,7 +244,11 @@ Ntuplizer::~Ntuplizer()
 void Ntuplizer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup){
   
   nBranches_->reset();
-      
+
+  edm::Handle<reco::VertexCollection> vertices;
+  iEvent.getByToken(vtxToken_, vertices);
+  if( vertices->empty() ) return; // skip the event if no PV found
```

Comment:
If there are no PV at all (size = 0), the whole event does not make much sense to me. This does not happen in MC but it does happen in data.